script.js

```diff
@@ -189,3 +217,112 @@ function sortDes() {
   });
   showRecipesInGrid();
 }
+
+//*************************** Set cooking timer ******************************/
+
+let startTime = 0;
+let timerId;
+
+const startBtn = document.getElementById("startBtn");
+startBtn.addEventListener("click", startBtnClicked);
+const pauseBtn = document.getElementById("pauseBtn");
+pauseBtn.addEventListener("click", pauseBtnClicked);
+
+function recipeCookingTimeClicked(cookingTime) {
+  stopTimer();
+  const timerShow = document.getElementById("timeContainer");
+  timerShow.style.display = "flex";
+
+  document.getElementById("timeInput").value = cookingTime;
+  const displayTime = document.getElementById("timeDisplay");
+  let hrs = 0;
+  let mins = 0;
+  hrs = Math.floor(cookingTime / 60);
+  if (hrs < 10) {
+    hrs = "0" + hrs;
+  }
+  mins = cookingTime % 60;
+  if (mins < 10) {
+    mins = "0" + mins;
+  }
+
+  displayTime.innerText = hrs + ":" + mins + ":00";
+}
+
+function startBtnClicked() {
+  startTime = document.getElementById("timeInput").value * 60;
+  document.getElementById("pauseBtn").innerHTML = "Pause";
+
+  startTimer();
+}
+
+function startTimer() {
+  timerId = setInterval(timerHandler, 1000);
+  startBtn.setAttribute("disabled", true);
+  pauseBtn.removeAttribute("disabled");
+  document.getElementById("timeInput").setAttribute("disabled", true);
+}
+
+function timerHandler() {
+  {
+    startTime--;
+
+    let hrs = Math.floor(startTime / 3600);
+
+    if (hrs < 10) {
+      hrs = "0" + hrs;
+    }
+    let mins = Math.floor((startTime % 3600) / 60);
+    if (mins < 10) {
+      mins = "0" + mins;
+    }
+    let secs = Math.floor((startTime % 3600) % 60);
+
+    if (secs < 10) {
+      secs = "0" + secs;
+    }
+
+    const displayTime = document.getElementById("timeDisplay");
+    displayTime.innerText = hrs + ":" + mins + ":" + secs;
+    if (startTime === 0) {
+      stopTimer();
+      playAlarm();
+    }
+  }
+}
+
+function pauseBtnClicked() {
+  if (document.getElementById("pauseBtn").innerHTML === "Pause") {
```

Comment:
It would be better to store the status of the timer in a variable rather than using the button's text.